README.md

```diff
@@ -1512,6 +1512,53 @@ would enable the use of a `@fold` on the `adjacent_animal` vertex field of `Foo`
 }
 ```
 
+## Schema Generation
+To generate a GraphQL schema, we first map a database schema into an intermediate representation:
+the `SchemaGraph`.
+
+#### SchemaGraph
+* `SchemaGraph`
+    * Keeps track of a schema's transitive inheritance structure.
+    * Keeps track of a schema's indexes. Indexes are not used in the GraphQL schema generation and 
+      have a provisional API.
+    * Represents a database's schema through `SchemaElement` objects.
+        * `SchemaElement`
+            * Can be one of three kinds:
+                * `VertexType`, which represent schema vertices.
+                * `EdgeType`, which represent relations between vertices.
+                * `NonGraphElement`, which represents a schema element with usually no connections
+                   to other classes. A `NonGraphElement` may only have connections if it's abstract 
+                   and all its non-abstract subclasses are `VertexTypes`. 
+            * Have the following attributes:
+                * `name`, indicating the name of the class.
+                * `abstract`, indicating whether the class is abstract or not.
+                * `properties`, which are represented using GraphQLTypes.
+                * `in_connections`/`out_connections`, which represent the set of possible adjacent 
```

Comment:
Give an example, otherwise this is mathematically correct and practically difficult to grasp.